blueflood-core/src/test/java/com/rackspacecloud/blueflood/types/BluefloodCounterRollupTest.java

```diff
@@ -1,93 +1,758 @@
-        Assert.assertNotSame(crData, crNoData);
-    }
-
-    private static <T> Points<T> asPoints(Class<T> type, long initialTime, long timeDelta, T... values) {
-        Points<T> points = new Points<T>();
-        long time = initialTime;
-        for (T v : values) {
-            points.add(new Points.Point<T>(time, v));
-            time += timeDelta;
-        }
-        return points;
-    }
-    
-    private static long[] makeRandomNumbers(int count) {
-        long[] numbers = new long[count];
-        for (int i = 0; i < count; i++)
-            numbers[i] = random.nextLong() % 1000000L;
-        return numbers;
-    }
-    
-    private static long sum(long[] numbers) {
-        long sum = 0;
-        for (long num : numbers)
-            sum += num;
-        return sum;
-    }
-    
-    // assume the samples are 30s apart.
-    private static BluefloodCounterRollup buildCounterRollupFromLongs(long... numbers) throws IOException {
-        long count = 0;
-        for (long number : numbers) {
-            count += number;
-        }
-        long sum = sum(numbers);
-        double rate = (double)sum / (double)(Constants.DEFAULT_SAMPLE_INTERVAL * numbers.length);
-        return new BluefloodCounterRollup().withCount(count).withRate(rate).withSampleCount(numbers.length);
+
+    @Test
+    public void equalsOtherNullReturnsFalse() {
+        //given
+        BluefloodCounterRollup rollup = new BluefloodCounterRollup();
+        //when
+        assertFalse(rollup.equals(null));
+    }
+
+    @Test
+    public void equalsOtherNotRollupReturnsFalse() {
+        //given
+        BluefloodCounterRollup rollup = new BluefloodCounterRollup();
+        //when
+        assertFalse(rollup.equals(""));
+    }
+
+    @Test
+    public void equalsDifferentCountReturnsFalse() {
+        //given
+        BluefloodCounterRollup a = new BluefloodCounterRollup().withCount(1);
+        BluefloodCounterRollup b = new BluefloodCounterRollup().withCount(2);
+        //when
+        assertFalse(a.equals(b));
+    }
+
+    @Test
+    public void equalsSameCountReturnsFalse() {
+        //given
+        BluefloodCounterRollup a = new BluefloodCounterRollup().withCount(1);
+        BluefloodCounterRollup b = new BluefloodCounterRollup().withCount(1);
+        //when
+        assertTrue(a.equals(b));
+    }
+
+    @Test
+    public void equalsDifferentRateReturnsFalse() {
+        //given
+        BluefloodCounterRollup a = new BluefloodCounterRollup().withCount(1).withRate(1.0d);
+        BluefloodCounterRollup b = new BluefloodCounterRollup().withCount(1).withRate(2.0d);
+        //when
+        assertFalse(a.equals(b));
+    }
+
+    @Test
+    public void equalsSameRateReturnsFalse() {
+        //given
+        BluefloodCounterRollup a = new BluefloodCounterRollup().withCount(1).withRate(1.0d);
+        BluefloodCounterRollup b = new BluefloodCounterRollup().withCount(1).withRate(1.0d);
+        //when
+        assertTrue(a.equals(b));
```

Comment:
@VinnyQ Ah, I see what you mean now. Yes, the name is misleading. Error on my part.